daffodil-runtime1-unparser/src/main/scala/org/apache/daffodil/processors/unparsers/ExpressionEvaluatingUnparsers.scala

```diff
@@ -103,13 +102,22 @@ class NewVariableInstanceStartUnparser(override val context: VariableRuntimeData
   override lazy val childProcessors = Vector()
 
   override def unparse(state: UState) = {
-    state.variableMap.newVariableInstance(context)
+    val v = state.variableMap.newVariableInstance(context)
+
     if (context.maybeDefaultValueExpr.isDefined) {
-      val maybeVar = state.variableMap.find(context.globalQName)
-      Assert.invariant(maybeVar.isDefined)
-      maybeVar.get.setState(VariableInProcess)
-      val suspendableExpression = new NewVariableInstanceSuspendableExpression(context.maybeDefaultValueExpr.get, context)
-      suspendableExpression.run(state)
+      val dve = context.maybeDefaultValueExpr.get
+      dve match {
+        case constExpr: ConstantExpression[_] => // Do nothing, value has already been computed at schema compile time
+        case _ => {
+          v.setState(VariableInProcess)
+          val suspendableExpression = new NewVariableInstanceDefaultValueSuspendableExpression(dve, context)
+          suspendableExpression.run(state)
+        }
+      }
+    } else {
+      // The NVI will inherit the default value of the original variable instance
+      // This will also inherit any externally provided bindings.
+      v.setDefaultValue(v.firstInstanceInitialValue)
```

Comment:
Just had a though, what if a variable did not have a default value and was not set externaly? Then firstInstanceInitialValue is NoValue, and so we are going to set the default value to not value. Which changes the state to Defined. So we could get in a state where we are Defined but with NoValue. Do we want to make it so firstInstanceInitialValue is an Option, and this becomes something like:
```scala
} else if (v.firstInstanceInitialValue.isDefined) {
  v.setDefaultValue(v.firstInstanceInitialValue.get)
}
```

Reply:
So we can actually just do a check to see if firstInstanceInitialValue is defined, which checks to see if it is something other than DataValue.NoValue.  So no need for making it an Option.

Reply:
Makes sense. I wonder if then it makes sense for _value to no longer be an Option then? Maybe ``def value`` and ``def stat`` just go away entirely if stateArg and valueArg aren't neeed anymore, and then ``value`` and ``state`` are just vars that are initlaized to NoValue and Undefined just like priorValue and priorState? Seems like _value only needed to be an option to deal with the pass by name of value, but I don't think we need that anymore?